tests/test_scenarios.py

```diff
@@ -21,12 +23,21 @@
 GIT_VERSION_RE = re.compile(r"(\d+\.)\d+(?:\.\d+)?(?:\.dev\d+\S+)?")
 
 
+def clean_output(output: str) -> str:
+    cleaned_output = LOG_LINE_TIMESTAMP_RE.sub("", output)
+    cleaned_output = LOG_LINE_NUMERIC_LINE_RE.sub(lambda m: (m.group(1) + "<n>"), cleaned_output)
+    cleaned_output = GIT_VERSION_RE.sub(lambda m: (m.group(1) + "<stuff>"), cleaned_output)
+    cleaned_output = re.sub(r"(?m) +$", "", cleaned_output)
```

Comment:
I didn't bring this up last time but I'm wary of `re.sub` with two strings; compiling it as a module constant also gives it a symbolic name like `END_OF_LINE_WHITESPACE`

Reply:
I'll make the change to fit the style. The other side of the coin: separating the pattern from the replacement makes it harder to see the complete operation in one place.  We can use keyword arguments if you want to distinguish pattern and repl.